language-plutus-core/examples/Language/PlutusCore/Examples/Data/TreeForest.hs

```diff
@@ -74,6 +74,10 @@ using this representation:
     Forest = AsForest TreeForest
 -}
 
+{- Note [Renaming]
+We do renaming in this module, because we normalize things and this requires renaming.
+-}
```

Comment:
So we do invoke type normalization not only from the type checker. Hence the normalizer should be split into two modules (internals, user-facing API) like the renamer and the type checker.

Reply:
Done.